Compilation on native Windows fails

### Logs

Steps to reproduce and logs :
[https://github.com/GauBen/bcotp/runs/1599361734?check_suite_focus=true#step:5:68](https://github.com/GauBen/bcotp/runs/1599361734?check_suite_focus=true#step:5:68)

```
# C:/cygwin/home/runneradmin/.opam/ocaml-variants.4.11.1+mingw64c/lib/ocaml/caml/mlvalues.h:265:24: warning: passing argument 1 of 'gr_font' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
#   265 | #define String_val(x) ((const char *) Bp_val(x))
#       |                       ~^~~~~~~~~~~~~~~~~~~~~~~~~
# src/win32/draw.c:340:17: note: in expansion of macro 'String_val'
# src/win32/draw.c:325:27: note: expected 'char *' but argument is of type 'const char *'
#   ocamlmklib src/dllgraphics_stubs.dll,src/libgraphics_stubs.a (exit 2)
# (cd _build/default && C:\cygwin\home\runneradmin\.opam\ocaml-variants.4.11.1+mingw64c\bin\ocamlmklib.opt.exe -g -o src/graphics_stubs src/draw.o src/events.o src/open.o -lkernel32 -lgdi32 -luser32)
# /usr/lib/gcc/x86_64-w64-mingw32/10/../../../../x86_64-w64-mingw32/bin/ld: C:\cygwin\tmp\dyndllf0869b.o:open.c:(.bss+0x5c): multiple definition of `grremember_mode'; C:\cygwin\tmp\dyndll23a1f4.o:draw.c:(.bss+0x60): first defined here
# /usr/lib/gcc/x86_64-w64-mingw32/10/../../../../x86_64-w64-mingw32/bin/ld: C:\cygwin\tmp\dyndllf0869b.o:open.c:(.bss+0x60): multiple definition of `grdisplay_mode'; C:\cygwin\tmp\dyndll23a1f4.o:draw.c:(.bss+0x64): first defined here
# collect2: error: ld returned 1 exit status
# ** Fatal error: Error during linking
```

Or:
* Install Cygwin
* Install Opam and OCaml 4.11.1
* opam install graphics

### Solution

I found a way to compile graphics on Cygwin:

* Remove or comment lines 26 and 27 in src/win32/draw.c
* Remove "conf-libX11" in dune-project and graphics.opam


Regards,
Gautier

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,6 +4,7 @@ Next
 - Fix configurator detection on native Windows (#19, @fdopen)
 - Use caml_alloc_custom_mem when available (#23, @hhugo)
 - Fix windows dependencies (#20, @diml)
+- Safe-string updates for native Windows (#28, fixes #27, @dra27)
 
 5.1.0 (05/12/2019)
 ------------------

--- a/src/win32/draw.c
+++ b/src/win32/draw.c
@@ -23,8 +23,6 @@
 #include "caml/version.h"
 
 HDC gcMetaFile;
-int grdisplay_mode;
-int grremember_mode;
 GR_WINDOW grwindow;
 
 static void GetCurrentPosition(HDC hDC,POINT *pt)

--- a/src/win32/draw.c
+++ b/src/win32/draw.c
@@ -323,7 +323,7 @@ CAMLprim value caml_gr_get_mousey(value unit)
 }
 
 
-static void gr_font(char *fontname)
+static void gr_font(const char *fontname)
 {
         HFONT hf = CreationFont(fontname);
 

--- a/src/win32/libgraph.h
+++ b/src/win32/libgraph.h
@@ -73,6 +73,6 @@ typedef struct tagWindow {
 } GR_WINDOW;
 
 extern GR_WINDOW grwindow;
-HFONT CreationFont(char *name);
+HFONT CreationFont(const char *name);
 extern void caml_gr_init_event_queue(void);
 extern void caml_gr_handle_event(UINT msg, WPARAM wParam, LPARAM lParam);

--- a/src/win32/open.c
+++ b/src/win32/open.c
@@ -42,7 +42,7 @@ static BOOL gr_initialized = 0;
 CAMLprim value caml_gr_clear_graph(value unit);
 HANDLE hInst;
 
-HFONT CreationFont(char *name)
+HFONT CreationFont(const char *name)
 {
    LOGFONTA CurrentFont;
    memset(&CurrentFont, 0, sizeof(LOGFONTA));

